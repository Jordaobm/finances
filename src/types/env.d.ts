declare module "@env" {
  export const WEBCLIENTID: string;
}

declare module "*.svg";
declare module "*.png";
declare module "*.jpg";
