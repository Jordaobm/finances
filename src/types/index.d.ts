declare module "*.png";
declare module "*.svg";
declare module "react-native-radial-gradient";
