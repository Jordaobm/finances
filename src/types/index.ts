export interface IconProps {
  color: string;
}

export interface HeaderProps {
  color: "white" | "black";
  onlySettings?: boolean;
}

export interface Operation {
  category: {
    color: string;
  };
  card: {
    name: string;
  };
  value: number;
  date: Date;
  name: string;
}

export interface CardOperationProps {
  operations: Operation[];
}

export interface NavigationProps {
  activeRoute: string;
}
