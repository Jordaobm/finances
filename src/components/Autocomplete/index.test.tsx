import React from "react";
import renderer from "react-test-renderer";
import { CustomAutocomplete } from ".";
import { AutocompleteOption } from "../../types";

jest.mock("react-native-autocomplete-input", () => "Autocomplete");

test("Component CustomAutocomplete", () => {
  const data: AutocompleteOption[] = [
    {
      label: "1",
      value: "1",
      id: "1",
    },
    {
      label: "2",
      value: "2",
      id: "2",
    },
  ];

  const tree = renderer
    .create(<CustomAutocomplete data={data} placeholder="Teste" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
