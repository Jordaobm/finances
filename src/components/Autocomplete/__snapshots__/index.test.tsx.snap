// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CustomAutocomplete 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "#f5f5f5",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 1,
        "flexDirection": "row",
        "paddingBottom": 4,
        "paddingLeft": 14,
        "paddingRight": 14,
        "paddingTop": 4,
        "position": "relative",
      },
    ]
  }
>
  <Autocomplete
    containerStyle={
      Object {
        "borderColor": "rgba(0,0,0,0)",
        "position": "relative",
        "width": "100%",
      }
    }
    data={Array []}
    flatListProps={
      Object {
        "keyExtractor": [Function],
        "keyboardShouldPersistTaps": "always",
        "renderItem": [Function],
      }
    }
    inputContainerStyle={
      Object {
        "borderColor": "rgba(0,0,0,0)",
      }
    }
    listContainerStyle={
      Object {
        "borderColor": "rgba(0,0,0,0)",
        "left": -5,
        "position": "absolute",
        "top": 40,
        "width": "100%",
      }
    }
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="Teste"
    placeholderTextColor="#c6c6c6"
    style={
      Array [
        Object {
          "borderColor": "white",
        },
        Object {
          "color": "#595959",
        },
      ]
    }
    value=""
  />
  <View
    style={
      Object {
        "paddingTop": 10,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24"
      bbWidth="24"
      fill="none"
      focusable={false}
      height="24"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width="24"
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M19.92 8.95L13.4 15.47C12.63 16.24 11.37 16.24 10.6 15.47L4.07996 8.95"
          propList={
            Array [
              "stroke",
            ]
          }
          stroke={4291217094}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
